controllers/gitrepository_controller.go

```diff
@@ -758,18 +759,18 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 	var gitReader git.RepositoryReader
 	var err error
 
-	switch obj.Spec.GitImplementation {
+	switch gitImplementation {
 	case sourcev1.LibGit2Implementation:
 		gitReader, err = libgit2.NewClient(dir, authOpts)
 	case sourcev1.GoGitImplementation:
 		gitReader, err = gogit.NewClient(dir, authOpts)
 	default:
-		err = fmt.Errorf("invalid Git implementation: %s", obj.Spec.GitImplementation)
+		err = fmt.Errorf("invalid Git implementation: %s", gitImplementation)
 	}
 	if err != nil {
 		// Do not return err as recovery without changes is impossible.
 		e := serror.NewStalling(
-			fmt.Errorf("failed to create Git client for implementation '%s': %w", obj.Spec.GitImplementation, err),
+			fmt.Errorf("failed to create Git client for implementation '%s': %w", gitImplementation, err),
```

Comment:
Not related to this PR, but while going through this, I noticed a potential issue above with the stalling error. Refer https://github.com/fluxcd/source-controller/pull/944/files#r1017795518 for the details.

Reply:
Changed with a generic error instead. PTAL